File: checkers/cwe/tests/suite/LoopLimitTaintChecker/java/incorrect_for_loop_struct_member3.java
import java.util.Scanner;
class A {
    int i;
}
class C {
    A a = new A();
}
class B {
    int main()
    {
        Scanner in = new Scanner(System.in);
        C c = new C();
        c.a.i = in.nextInt();
        for (int i = 0; i < c.a.i; i++) {  // Loop here
        }
    }
}

File: checkers/cwe/tests/suite/LoopLimitTaintChecker/incorrect_for_loop_struct_member3.cpp
#include <cstdlib>
#include <fstream>
struct B {
    int i;
};
struct A {
    B* b;
};

void f1()
{
    A* a;
    std::ifstream ifs("file");
    ifs >> a->b->i;

    for (int i = 0; i < a->b->i; i++) {  // Loop here
    }
}
